runelite-client/src/main/java/net/runelite/client/plugins/weaponcharges/WeaponChargesPlugin.java

```diff
@@ -0,0 +1,276 @@
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import java.awt.Color;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Actor;
+import net.runelite.api.AnimationID;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.api.EquipmentInventorySlot;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.events.AnimationChanged;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.ItemContainerChanged;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+@PluginDescriptor(
+	name = "Weapon Charges"
+)
+@Slf4j
+public class WeaponChargesPlugin extends Plugin
+{
+	private static final int WEAPON_SLOT = EquipmentInventorySlot.WEAPON.getSlotIdx();
+	private static final Pattern ibansCheckString = Pattern.compile("You have ([0-9]+) charges left on the staff.");
+	private static final String ibansRechargeString = "You hold the staff above the well...";
+
+	private WeaponChargesCounter counter;
+
+	private ChargedWeapon rechargedWeapon;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private WeaponChargesOverlay overlay;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private WeaponChargesConfig config;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Provides
+	WeaponChargesConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(WeaponChargesConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		processChargesCounter();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		removeChargesCounter();
+	}
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	Integer getCharges(ChargedWeapon weapon)
+	{
+		if ((weapon == ChargedWeapon.IBANS_STAFF ||
+				weapon == ChargedWeapon.IBANS_STAFF_U) &&
+				config.ibansCharges() >= 0)
+		{
+			return config.ibansCharges();
+		}
+		else
+		{
+			return null;
+		}
+	}
+
+	Color getChargesColor(Integer charges)
+	{
+		if (charges == null)
+		{
+			return Color.MAGENTA;
+		}
+		else if (charges > 100)
+		{
+			return Color.WHITE;
+		}
+		else
+		{
+			return Color.RED;
+		}
+	}
+
+	private void addChargesCounter(ChargedWeapon weapon)
+	{
+		if (counter != null)
+		{
+			if (counter.getChargedWeapon() == weapon)
+			{
+				return;
+			}
+
+			removeChargesCounter();
+		}
+
+		counter = new WeaponChargesCounter(itemManager.getImage(weapon.getItemId()), this, weapon);
```

Comment:
Run with `-ea` and this should assert. You can't `getImage` outside of the client thread.